transport/src/main/java/io/netty/channel/DefaultChannelHandlerInvoker.java

```diff
@@ -506,13 +419,20 @@ private static WriteTask newInstance(
             task.ctx = ctx;
             task.msg = msg;
             task.promise = promise;
-            task.size = ((AbstractChannel) ctx.channel()).estimatorHandle().size(msg)
-                    + estimateSize(task, CLASS_SIZES.get());
-            ChannelOutboundBuffer buffer = ctx.channel().unsafe().outboundBuffer();
-            // Check for null as it may be set to null if the channel is closed already
-            if (buffer != null) {
-                buffer.incrementPendingOutboundBytes(task.size);
+
+            if (ESTIMATE_TASK_SIZE_ON_SUBMIT) {
+                task.size = ((AbstractChannel) ctx.channel()).estimatorHandle().size(msg) + WRITE_TASK_OVERHEAD;
```

Comment:
@technocoreai just one nit.. can you move this in the `if (buffer != null) { ... }` block and if the buffer == null set the size = 0 ?